engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestrator.java

```diff
@@ -3018,13 +3018,8 @@ protected void checkL2OfferingServices(NetworkOfferingVO ntwkOff) {
     @DB
     public boolean shutdownNetwork(final long networkId, final ReservationContext context, final boolean cleanupElements) {
         NetworkVO network = _networksDao.findById(networkId);
-        if (network.getState() == Network.State.Allocated) {
-            s_logger.debug("Network is already shutdown: " + network);
-            return true;
-        }
-
-        if (network.getState() != Network.State.Implemented && network.getState() != Network.State.Shutdown) {
-            s_logger.debug("Network is not implemented: " + network);
+        if (network == null) {
+            s_logger.debug("Network with id: " + networkId + " doesn't exists");
```

Comment:
seems to me this part is redundant as well as the null check is the first thing done after getting the lock, @sureshanaparti ?

Reply:
@DaanHoogland acquire lock can return null if lock fails or network doesn't exists. this check avoids locking call.

Reply:
ok, but it adds the need for the findById call. no big issue either way

Reply:
ok, updated the log and removed findById call @DaanHoogland